File: solutions/1614. Maximum Nesting Depth of the Parentheses/1614.cpp
class Solution {
 public:
  int maxDepth(string s) {
    int ans = 0;
    int opened = 0;

    for (const char c : s)
      if (c == '(')
        ans = max(ans, ++opened);
      else if (c == ')')
        --opened;

    return ans;
  }
};

File: solutions/1614. Maximum Nesting Depth of the Parentheses/1614.py
class Solution:
  def maxDepth(self, s: str) -> int:
    ans = 0
    opened = 0

    for c in s:
      if c == '(':
        opened += 1
        ans = max(ans, opened)
      elif c == ')':
        opened -= 1

    return ans
